src/lib.rs

```diff
@@ -1329,18 +1484,39 @@ impl Bump {
             let mut base_size = (current_layout.size() - FOOTER_SIZE)
                 .checked_mul(2)?
                 .max(min_new_chunk_size);
-            let sizes = iter::from_fn(|| {
-                if base_size >= min_new_chunk_size {
+            let chunk_memory_details = iter::from_fn(|| {
+                let bypass_min_chunk_size_for_small_limits = match self.allocation_limit() {
+                    Some(limit)
+                        if layout.size() < limit
+                            && base_size >= layout.size()
+                            && limit < DEFAULT_CHUNK_SIZE_WITHOUT_FOOTER
+                            && self.allocated_bytes() == 0 =>
+                    {
+                        true
+                    }
+                    _ => false,
+                };
+
+                if base_size >= min_new_chunk_size || bypass_min_chunk_size_for_small_limits {
                     let size = base_size;
                     base_size = base_size / 2;
-                    Some(size)
+                    Bump::new_chunk_memory_details(Some(size), layout)
```

Comment:
I'm not sure if rustc is smart enough to transform this into
```rust
if base_size >= min_new_chunk_size {
    // ...
} else {
  let bypass_min_chunk_size_for_small_limits = ...;
  if bypass_min_chunk_size_for_small_limits {
     let size = base_size;
     base_size = base_size / 2;
     Bump::new_chunk_memory_details(Some(size), layout)
  } else {
    None
  }
}
```
but I decided to hold off on any premature optimizations since I think this is much less readable